docs/book/src/faq.md

```diff
@@ -92,6 +92,29 @@ securityContext:
 ```
 However, note that this problem is fixed and will not occur if you deploy the project in high versions (maybe >= 1.22).
 
+## The error `Too long: must have at most 262144 bytes` is faced when I run `make install` to apply the CRD manifests . How to solve it? Why this error is faced?
+
+The Kubernetes API has a size limit. Then, when the apply is used the API will add the annotation last-applied-configuration with all previous configuration which is exceeding this amount. [More info](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/declarative-config/#how-to-create-objects)
+
+Therefore, you have a few options to workround this scenario such as:
+
+- You can use the option `maxDescLen=0` to remove the description of your CRDs. Following an example:
+
+```shell
+
+ .PHONY: manifests 
+ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects. 
+     # Note that the option maxDescLen=0 was added in the default scaffold in order to sort out the issue 
+     # Too long: must have at most 262144 bytes. By using kubectl apply to create / update resources an annotation 
+     # is created by K8s API to store the latest version of the resource ( kubectl.kubernetes.io/last-applied-configuration). 
+     # However, it has a size limit and if the CRD is too big with so many long descriptions as this one it will cause the failure. 
+ 	$(CONTROLLER_GEN) rbac:roleName=manager-role crd:maxDescLen=0 webhook paths="./..." output:crd:artifacts:config=config/crd/bases 
+
+Instead of Kubectl create command use Kubectl apply and it will work because kubectl create will send the request at once and it will fail while kubect apply send the request in small chunks so the desc len will be divided. 
+
+Add maxDescLen=0  and it will ensure that the crd description size is utmost and hence it will go through it. 
+
```

Comment:
Note that in my comment, I added one more workaround option to ensure that the Single Responsibility principle is not violated. Can you please check that?

Also, we need to mention the ideal solution here, which would be to work with server-side, but it is not yet supported. In this case, would be either helpful ensure that we link the PR for controller-gen (please see my comment with the suggestions which has the PR as the issue)

Reply:
I have added the link of controller-tools pr. But I am unsure about **_Single Responsibility principle_** In this [comment](https://github.com/kubernetes-sigs/kubebuilder/issues/3460#issuecomment-1635365658) You have talked about DDD principle. and in whole pr I didnt find anything about Single Responsibility principle. Can you please explain?